### ISSUE TITLE
no way to resolve sync conflict

### ISSUE THREAD

**Issue by samdmarshall** [mssun/passforios#392]
i'm having a problem where performing a sync operation results in a merge conflict, but due to the way the git library is wrapped and no file-system access within the app; i cannot resolve this without permanent data-loss.

can there please be a diagnositic option added to the advanced settings that either: 
* allows me to export a diff of all my local changes too clipboard or saved as a file 
* adds support for pass as a file-system provider (so that the contents of the repo can be browsed with the files app and manually get exported)
* allow another app to provide the .password-store directory instead, so an app that is equipped to actually perform git operations is handling them (like working copy)

my password store on my phone is out of sync for over a year now because of a minor conflict this app cannot resolve on it's own, and it is too risky/impossible to save or export the local copy of the password store.

**Comment by SimplyDanny** [comment]
You can always resolve merge conflicts on a device which allows to execute Git commands. Clone the repository, backup the current state, reset the branch synced with Pass for iOS to a state before any changes in the app happened, add your changes in the app to the branch, apply your locally saved changes and sync again.

**Comment by samdmarshall** [comment]
@SimplyDanny if it were possible to do that, i wouldn't have had to make this issue now would i?

**Comment by SimplyDanny** [comment]
Well, you may have your reasons. For me it is not clear what your actual issue is. You should rather describe your problem instead of proposing a possible solution.

**Comment by tarak** [comment]
"no file-system access within the app"...

How to perform git commands on an iOS device without being app developer, maybe?

The question is how to not lose passwords created on a device that are not pushed to the GIT repo due to a merge conflict, if I understand correctly?

What are the options when such problems arise?

I had different merge conflicts due to my own stupidity on MacBooks (with GIT installed) where I could solve these issues using my terminal, but what can one do on an iOS device to solve merge conflicts to resume sync other than deleting all data in the app (and with it the commits or passwords that were not pushed to the GIT repo)?

**Comment by jmsundar** [comment]
>backup the current state

@SimplyDanny could you explain how I can backup the current state of the repository?

**Comment by jmsundar** [comment]
> 
> 
> "no file-system access within the app"...
> 
> How to perform git commands on an iOS device without being app developer, maybe?
> 
> The question is how to not lose passwords created on a device that are not pushed to the GIT repo due to a merge conflict, if I understand correctly?
> 
> What are the options when such problems arise?
> 
> I had different merge conflicts due to my own stupidity on MacBooks (with GIT installed) where I could solve these issues using my terminal, but what can one do on an iOS device to solve merge conflicts to resume sync other than deleting all data in the app (and with it the commits or passwords that were not pushed to the GIT repo)?

https://github.com/mssun/passforios/issues/131#issuecomment-442642737

You should be able to use the process described in this comment to resolve this.

**Comment by SimplyDanny** [comment]
The [comment](https://github.com/mssun/passforios/issues/131#issuecomment-442642737) you mentioned is exactly the workaround I had in mind when I wrote me first comment in this thread. 

Depending on the overall setup, one device with access to the Git command line is usually sufficient to resolve conflicts. I see no reason why the app (a password manager) should offer all this functionality.

**Comment by samdmarshall** [comment]
i ended up being able to resolve this but it was a terrible experience -- the vast majority of this issue comes from git being a horrific piece of software.

to redirect the conversation to the actual issue: this app needs to provide a way to export current local data. i don't care if it is as a patch file, or a zip archive of the repo directory or whatever, but right now this app is a black box and that isn't helpful when trying to resolve any sync issue.

**Comment by StarrWulfe** [comment]
I agree. There at least should be a way to do a `x-callback-URL` workflow with something like Working Copy.

**Comment by samdmarshall** [comment]
i spotted this other issue #460 which sounds like it is fundamentally asking the same thing, eg: being able to access the contents of the local password repo, in a way that is needlessly complicated by the underlying reliance on git. in my case i want to examine the difference between my local and remote commits, whereas in the other thread, they just want to be able to push and pull their contents to another git origin because that would be the only way to work around local data loss when it all doesn't "just work". i'm going to update my original issue to try to better reflect what is desired here.
